src/lib.rs

```diff
@@ -190,6 +194,31 @@ pub trait Actor: 'static + Send + Sized {
     }
 }
 
+/// An error related to the actor system
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub enum Error {
+    /// The actor is no longer running and disconnected from the sending address.
+    Disconnected,
```

Comment:
For encapsulation purposes, we could make this enum private and encapsulating it in an opaque error, similar to `std::io::Error` but we'd lose exhaustive matching then because users can only check of the types via `Error::is_interrupted()` etc.

Reply:
I think that having exhaustive matching is probably useful, so I would rather leave it as it is now, unless there is a compelling reason to encapsulate it

Reply:
Also, should the interrupted variant also have the name?

Reply:
> Also, should the interrupted variant also have the name?

I guess so? The name is useful because this error may often be question-marked and then appear in an error chain from `anyhow` or something. Without the actor name, it is really hard to debug where it came from.

Reply:
In that case, I wonder if it should be extracted as a field:

```rust
struct Error {
    act: &'static str,
    kind: ErrorKind,
}
```
with the current `Error` renamed to `ErrorKind`

Reply:
I've been working on this and it's kinda outgrown this PR, so would you be ok with merging this now and having this feature in a followup?